packages/js-drive/lib/abci/handlers/infoHandlerFactory.js

```diff
@@ -48,24 +45,17 @@ function infoHandlerFactory(
 
     // Initialize Block Execution Contexts
 
-    const persistedBlockExecutionContextStack = await blockExecutionContextStackRepository.fetch();
-
-    blockExecutionContextStack.setContexts(persistedBlockExecutionContextStack.getContexts());
-
-    const previousContext = blockExecutionContextStack.getFirst();
-
-    // Populate current execution context with previous context
-    // until block begin called with a new block.
-    if (previousContext) {
-      blockExecutionContext.populate(previousContext);
+    const persistedBlockExecutionContext = await blockExecutionContextRepository.fetch();
+    if (persistedBlockExecutionContext) {
+      blockExecutionContext.populate(persistedBlockExecutionContext);
```

Comment:
Let's rename `blockExecutionContext` in container to `latestBlockExecutionContext`